core/src/main/java/cucumber/runtime/Runtime.java

```diff
@@ -121,6 +121,12 @@ public void runFeature(CucumberFeature feature) {
         }
     }
 
+    public void runPickle(PickleEvent pickle) {
+        if (matchesFilters(pickle)) {
```

Comment:
Already the data provider `provideScenarios` should filter the `PickleEvent`, otherwise it will look like they were run and passed, when in fact they were not selected for execution.